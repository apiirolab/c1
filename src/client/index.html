<!DOCTYPE html>
<html>

<head>
  <title></title>
  <meta name="description" content="Corporate process management for Western Sydney University!">
  <link rel="icon" type="image/png" href="/favico.png">
  <link href='https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.3.2/semantic.min.css' rel='stylesheet' type='text/css'>
</head>

<body>
  <div id="root"></div>
  $bundles
</body>

</html>