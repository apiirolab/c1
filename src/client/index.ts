export const info = 'Client Info';
