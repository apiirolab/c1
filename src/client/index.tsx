import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { ApolloProvider } from 'react-apollo';
import { Routes } from './config/router';
import { client } from './config/apollo';
// import * as serviceWorker from './registerServiceWorker';

ReactDOM.render(
  <ApolloProvider client={client}>
    <Routes />
  </ApolloProvider>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
// serviceWorker.unregister();
