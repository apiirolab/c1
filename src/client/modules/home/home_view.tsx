import * as React from 'react';

import { inject, observer } from 'mobx-react';
import { Header, Segment } from 'semantic-ui-react';
import styled from 'styled-components';

import { MainLayout } from '../core/main_layout';
import { HeaderAnonymous } from '../headers/header_anonymous';
import { LoginContainer } from '../login/login_container';

const Centered = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

declare global {
  interface PageProps {
    path?: string;
  }
}

export const HomeLayout: React.SFC<PageProps> = () => (
  <MainLayout>
    <HomeView />
  </MainLayout>
);

type Props = {
  store?: App.Store;
  path?: string;
};

const LoginSegment = styled(Segment)`
  width: 450px;
  margin: auto;
`;

export const HomeViewAuth = () => <div />;

export const HomeViewAnonymous = () => (
  <Centered>
    <div style={{ width: '400px' }}>
      <LoginSegment secondary>
        <Header icon="lock" content="Login to CORPIX" dividing />
        <LoginContainer />
      </LoginSegment>
    </div>
  </Centered>
);

export const HomeView: React.SFC<Props> = ({ store }) => (
  <>
    <Choose>
      <When condition={!store.userId}>
        <>
          <HeaderAnonymous />
          <HomeViewAnonymous />
        </>
      </When>
      <Otherwise>
        <MainLayout>
          <div>Home</div>
        </MainLayout>
      </Otherwise>
    </Choose>
  </>
);

export const HomeContainer = inject('store')(observer(HomeView));
