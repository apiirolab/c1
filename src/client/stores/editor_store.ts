import { types } from 'mobx-state-tree';

export const EditorStore = types
  .model('EditorStore', {
    dirty: types.maybe(types.boolean)
  })
  .actions(self => ({
    setDirty(value: boolean) {
      self.dirty = value;
    }
  }));

export const editorStore = EditorStore.create({});
