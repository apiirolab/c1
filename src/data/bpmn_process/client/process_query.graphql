query ProcessQuery($id: ID!) {
  process(id: $id) {
    actionCount
    description
    id
    model
    resources {
      name
      type
    }
    status
    version
  }
}
