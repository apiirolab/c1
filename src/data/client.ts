import * as QueryTypes from './generated/types';

export { default as gql } from 'graphql-tag';

export { QueryTypes };

export function purge<T>(params: any): T {
  for (let key of Object.getOwnPropertyNames(params)) {
    if (!params[key]) {
      delete params[key];
    }
    if (params[key] instanceof Object) {
      purge(params[key]);
    }
  }
  return params;
}
