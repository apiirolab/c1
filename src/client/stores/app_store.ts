import i18n from 'es2015-i18n-tag';
import { types } from 'mobx-state-tree';

import { client } from 'client/config/apollo';
import { QueryTypes } from 'data/client';
import { LocalStorage } from '../config/local_storage';
import { LoginStore } from '../modules/login/login_store';
import { UserStore } from '../modules/user/user_store';

declare global {
  namespace App { type Store = typeof AppStore.Type; }
}

export const AppStore = types
  .model('AppStore', {
    userId: types.maybe(types.string),
    user: types.maybe(UserStore),
    login: types.optional(LoginStore, () => LoginStore.create({}))
  })
  .volatile(_self => ({
    i18n,
    client,
    localStorage: LocalStorage
  }))
  .actions(self => {
    return {
      setUser(user: QueryTypes.User) {
        self.login.password = '';
        self.userId = user.id;
        self.user = user as any;
        self.localStorage.userId = user.id;
      },
      logout() {
        self.userId = undefined;
        self.user = undefined;
        self.client().resetStore();
        self.localStorage.userId = null;
        self.localStorage.token = null;
      }
    };
  });

export const appStore = AppStore.create({ userId: LocalStorage.userId });
