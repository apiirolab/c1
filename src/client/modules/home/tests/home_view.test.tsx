import * as React from 'react';

import { Provider } from 'mobx-react';
import { create } from 'react-test-renderer';

import { AppStore } from 'client/stores/app_store';
import { MockedProvider } from 'tests/client';
import { HomeView } from '../home_view';

describe('Home', () => {
  function componentWithData() {
    const store = AppStore.create({});

    return (
      <Provider store={store}>
        <MockedProvider>
          <HomeView store={store} />
        </MockedProvider>
      </Provider>
    );
  }

  describe('Logged Out', () => {
    it('renders for logged out user', () => {
      const root = create(componentWithData());
      expect(root).toMatchSnapshot();
    });

    return { componentWithData };
  });

  describe('Logged In', () => {
    it('renders for logged in user', () => {
      const root = create(componentWithData());
      expect(root).toMatchSnapshot();
    });

    return { component: componentWithData() };
  });
});
