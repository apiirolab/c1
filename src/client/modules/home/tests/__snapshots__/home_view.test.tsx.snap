// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Logged In renders for logged in user 1`] = `
<div
  className="sc-iwsKbI kahTAg"
>
  <div
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="ui secondary segment sc-gZMcBi gQPKdT"
    >
      <div
        className="ui dividing header"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
        <div
          className="content"
        >
          Login to CORPIX
        </div>
      </div>
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label>
            User
          </label>
          <div
            className="ui fluid icon input"
          >
            <input
              autoComplete="username"
              name="user"
              onChange={[Function]}
              placeholder="Your Staff ID"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
            />
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Password
          </label>
          <div
            className="ui fluid icon input"
          >
            <input
              autoComplete="current-password"
              name="password"
              onChange={[Function]}
              placeholder="Your Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <button
            className="ui primary button sc-htoDjs dhyhDz"
            onClick={[Function]}
            role="button"
          >
            <i
              aria-hidden="true"
              className="thumbs up icon"
            />
            Login
          </button>
          <a
            className="sc-dnqmqq bFBsXy"
            href="https://www.westernsydney.edu.au/information_technology_services/its/servicedesk/password_expiry_faq"
            target="_blank"
          >
            Forgot Password?
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Home Logged Out renders for logged out user 1`] = `
<div
  className="sc-iwsKbI kahTAg"
>
  <div
    style={
      Object {
        "width": "400px",
      }
    }
  >
    <div
      className="ui secondary segment sc-gZMcBi gQPKdT"
    >
      <div
        className="ui dividing header"
      >
        <i
          aria-hidden="true"
          className="lock icon"
        />
        <div
          className="content"
        >
          Login to CORPIX
        </div>
      </div>
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label>
            User
          </label>
          <div
            className="ui fluid icon input"
          >
            <input
              autoComplete="username"
              name="user"
              onChange={[Function]}
              placeholder="Your Staff ID"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
            />
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Password
          </label>
          <div
            className="ui fluid icon input"
          >
            <input
              autoComplete="current-password"
              name="password"
              onChange={[Function]}
              placeholder="Your Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <button
            className="ui primary button sc-htoDjs dhyhDz"
            onClick={[Function]}
            role="button"
          >
            <i
              aria-hidden="true"
              className="thumbs up icon"
            />
            Login
          </button>
          <a
            className="sc-dnqmqq bFBsXy"
            href="https://www.westernsydney.edu.au/information_technology_services/its/servicedesk/password_expiry_faq"
            target="_blank"
          >
            Forgot Password?
          </a>
        </div>
      </form>
    </div>
  </div>
</div>
`;
